### PR TITLE
Redesign upload testdata generation

### PR DESCRIPTION
**Pull request by fredrb**
## Checklist

* [x] Test your changes as thoroughly as possible before you commit them. Preferably, automate your test by unit/integration tests.
* [x] Make sure `mvn install` runs for the whole project and, if you touched any code in the respective service, submission and distribution service can be run with `spring-boot:run`

## Description

Reworks a bit on our Test Data Generation for Upload. Since we migrated the policies from distribution (i.e. key expiry check) our current mechanism was generating key that would be filtered out from Upload. With that in mind, we changed the following things:

1. Generate keys for day before
2. Only generate keys from latest generated key (avoid always generating a lot of keys every upload run)
3. Change the number test keys property from `keys` to `keys-per-hour` to reflect how the generation is being done. Other than having a big set of keys (e.g. 10k) and try to fit them in a 24h window (by splitting in keys per hours), it generates a smaller set of keys (e.g. 350) for each free hour from yesterday.

### CODE REVIEW

File: services/upload/src/main/java/app/coronawarn/server/services/federation/upload/runner/TestDataGeneration.java [hunk 1]
```diff
@@ -67,8 +72,8 @@ public TestDataGeneration(UploadServiceConfig uploadServiceConfig,
     return keydata;
   }
 
-  private DiagnosisKey makeKeyFromTimestamp(long timestamp) {
-    return DiagnosisKey.builder().withKeyData(randomByteData())
+  private FederationUploadKey makeKeyFromTimestamp(long timestamp) {
+    return FederationUploadKey.from(DiagnosisKey.builder().withKeyData(randomByteData())
         .withRollingStartIntervalNumber(generateRollingStartIntervalNumber(timestamp))
         .withTransmissionRiskLevel(2)
```
Comment (hilmarf) [inline services/upload/src/main/java/app/coronawarn/server/services/federation/upload/runner/TestDataGeneration.java:78]: fixed value of `2`?

File: services/upload/src/main/java/app/coronawarn/server/services/federation/upload/runner/TestDataGeneration.java [hunk 2]
```diff
@@ -77,7 +82,7 @@ private DiagnosisKey makeKeyFromTimestamp(long timestamp) {
         .withSubmissionTimestamp(timestamp)
         .withVisitedCountries(List.of("FR", "DK"))
```
Comment (hilmarf) [inline services/upload/src/main/java/app/coronawarn/server/services/federation/upload/runner/TestDataGeneration.java:83]: shouldn't the list of countries be configurable or use the list of available countries?

**Review by hilmarf — DISMISSED** [review]

File: services/upload/src/main/java/app/coronawarn/server/services/federation/upload/runner/TestDataGeneration.java (hunk 2 above)
Comment (fredrb) [reply services/upload/src/main/java/app/coronawarn/server/services/federation/upload/runner/TestDataGeneration.java:83]: The configuration for supported countries was done in distribution and submission, but not in upload. So we don't have this information here.

**Review by hilmarf — APPROVED** [review]

File: services/upload/src/main/java/app/coronawarn/server/services/federation/upload/runner/TestDataGeneration.java [hunk 3]
```diff
@@ -67,17 +72,22 @@ public TestDataGeneration(UploadServiceConfig uploadServiceConfig,
...
         .withReportType(ReportType.CONFIRMED_TEST)
-        .build();
+        .build());
+  }
+
+  private int generateTransmissionRiskLevel() {
+    return Math.toIntExact(
+        getRandomBetween(RiskLevel.RISK_LEVEL_LOWEST_VALUE, RiskLevel.RISK_LEVEL_HIGHEST_VALUE));
```
Comment (EugenM-SAP) [inline services/upload/src/main/java/app/coronawarn/server/services/federation/upload/runner/TestDataGeneration.java:90]: Not sure whether it's an issue or not that TRL values like 2 or 4 might be produced. Considering that the german app does not submit such values. Probably for testing it's not a problem

**Review by EugenM-SAP — APPROVED** [review]

File: services/upload/src/main/java/app/coronawarn/server/services/federation/upload/runner/TestDataGeneration.java (hunk 3 above)
Comment (fredrb) [reply services/upload/src/main/java/app/coronawarn/server/services/federation/upload/runner/TestDataGeneration.java:90]: Agreed. Maybe we have to think this through in the future. But distribution does it like this also. DEV and INT are filled with keys that have the wrong TRL.

**Pull request merged**